cli/command_snapshot_create.go

```diff
@@ -470,3 +447,62 @@ func shouldSnapshotSource(ctx context.Context, src snapshot.SourceInfo, rep repo
 		src.UserName == rep.ClientOptions().Username &&
 		!policy.IsManualSnapshot(policyTree), nil
 }
+
+func (c *commandSnapshotCreate) getContentToSnapshot(ctx context.Context, dir string, rep repo.RepositoryWriter) (fs.Entry, snapshot.SourceInfo, bool, error) {
+	var (
+		absDir     string
+		sourceInfo snapshot.SourceInfo
+		fsEntry    fs.Entry
+		setManual  bool
+		err        error
+	)
+
+	absDir, err = filepath.Abs(dir)
+	if err != nil {
+		return nil, sourceInfo, false, errors.Wrapf(err, "invalid source %v", dir)
+	}
+
+	if c.sourceOverride != "" {
+		sourceInfo, err = parseFullSource(c.sourceOverride, rep)
```

Comment:
nit: change the function parameter to only require either `rep.ClientOptions` or `hostname, username string`